ide/coqide/ideutils.ml

```diff
@@ -179,14 +179,14 @@ let byte_off_to_buffer_off buffer byte_off =
   cvt (buffer#get_iter `START) byte_off
 
 (** convert unicode offset (used by GTK buffer) to UTF-8 byte offset (used by Coq) *)
-let buffer_off_to_byte_off (buffer : GText.buffer) uni_off =
+let buffer_off_to_byte_off (buffer : GText.buffer) (s_byte, s_uni) uni_off =
   let rec cvt iter rv =
-    if iter#offset <= 0 then
+    if iter#offset <= s_uni then
```

Comment:
This is incorrect if the function is called on a range that occurs before the cache.

Reply:
I assumed the sentences would be processed in ascending order.  Is that not guaranteed?

Reply:
Incorrect meaning slow.

Reply:
In all its current uses, the function is correct. But this is part of a public API, and if it happens to be used without this precondition it will produce garbage results.

Reply:
I don't see it.  If the value is (0,0) then it goes through the entire string.  Slower than we want but correct.

Reply:
The bug happens if you put it to something else than (0, 0). It will probably loop (not sure, depends what GTK does when iterating before the buffer start).

Reply:
(The simplest way to fix this is to keep this function as-is there but redefine the one using the cache in the scriptview module.)

Reply:
Done.